trio_websocket/_impl.py

```diff
@@ -181,10 +181,27 @@ async def _close_connection(connection: WebSocketConnection) -> None:
         except trio.TooSlowError:
             raise DisconnectionTimeout from None
 
+    def _raise(exc: BaseException) -> NoReturn:
+        """This helper allows re-raising an exception without __context__ being set."""
+        # cause does not need special handlng, we simply avoid using `raise .. from ..`
+        __tracebackhide__ = True
+        context = exc.__context__
+        try:
+            raise exc
+        finally:
+            exc.__context__ = context
+            del exc, context
```

Comment:
just remembered an issue with this:

https://github.com/python/cpython/issues/99856
https://github.com/python-attrs/attrs/pull/1081

```suggestion
        context = exc.__context__
        try:
            raise exc
        finally:
            object.__setattr__(exc, "__context__", context)
            del exc, context
```

Reply:
You'll probably need a test with a frozen dataclass and attrs class Exception

Reply:
I can fix them... but `open_websocket` is decorated with `@asynccontextmanager` so it just kicks the can onto failing in the same way as https://github.com/python/cpython/issues/99856

For `attrs` compatibility I kind of feel like we can require users with frozen exception to require `trio_websocket<=0.11.1` or `attrs>24.2.0` (both to be released) - or change their exceptions. It's not a heavy workaround for us, but I still don't really think it's worth it.